js/focuspoint/jquery.focusPoint.js

```diff
@@ -0,0 +1,436 @@
+			{
+				base.focusInterface.updateStylePosition();
+				base.focusInterface.updateStyleBackground();
+			},
+
+			updateStylePosition: function ()
+			{
+				base.focusInterface.$el.css({
+					left: base.attachment._focusPoint.x + '%',
+					top: base.attachment._focusPoint.y + '%'
+				});
+			},
+
+			updateStyleBackground: function ()
+			{
+				var posX = 0 - (base.focusInterface._position.x - base.focusInterface._radius);
+				var posY = 0 - (base.focusInterface._position.y - base.focusInterface._radius);
+
+				base.focusInterface.$el.css({
+					backgroundImage: 'url("' + base.attachment.$el.attr('src') + '")',
+					backgroundSize: base.attachment._width + 'px ' + base.attachment._height + 'px ',
+					backgroundPosition: posX + 'px ' + posY + 'px '
+				});
+			},
+
+			/**
+			 * Calculation click position within the focusInterface for focalpoint calculation
+			 *
+			 * @param event
+			 */
+			updateClickPosition: function (event)
+			{
+				base.focusInterface._clickPosition.x = event.pageX - base.focusInterface._offset.x;
+				base.focusInterface._clickPosition.y = event.pageY - base.focusInterface._offset.y;
+			},
+
+			/**
+			 * Update dimension data of the focusInterface for quick access to the latest dimensions
+			 */
+			updateDimensionData: function ()
+			{
+				// Get width and height in pixels
+				base.focusInterface._width = base.focusInterface.$el.width();
+				base.focusInterface._height = base.focusInterface.$el.height();
+
+				// Calculate the radius in px of the focusInterface based on width
+				base.focusInterface._radius = base.focusInterface._width / 2;
+
+				// Write offset based on the center point of the focusInterface
+				base.focusInterface._offset.x = base.focusInterface.$el.offset().left + base.focusInterface._radius;
+				base.focusInterface._offset.y = base.focusInterface.$el.offset().top + base.focusInterface._radius;
+
+				// Write position based on the calculation position of focuspoint of the attachment
+				base.focusInterface._position.x = (base.attachment._focusPoint.x / 100) * base.attachment._width;
+				base.focusInterface._position.y = (base.attachment._focusPoint.y / 100) * base.attachment._height;
+			}
+		};
+
+		/**
+		 *  Crop Button functions
+		 */
+		base.cropButton = {
+			$el: false,
+
+			init: function ()
+			{
+				var button = '<button type="button" class="' + css.button.self + ' ' + css.button.disabled + ' crop-attachment ' + css.imageFocus.button + '">' + focusPointL10n.cropButton + '</button>';
+				$(base.el).find('.attachment-actions').append(button);
+
+				base.cropButton.$el = $('.' + css.imageFocus.button);
+
+				//Set action to button for ajax call
+				base.cropButton.$el.on('click', base.sendImageCropDataByAjax);
+			},
+			highlight: function ()
+			{
+				base.cropButton.$el.removeClass(css.button.disabled);
+				base.cropButton.$el.text(focusPointL10n.cropButton);
+				base.cropButton.$el.addClass(css.button.primary);
+			},
+			activate: function ()
```

Comment:
I think it makes no sense yet to take this kind of improvements, the code is gonna change constantly in the beginning until we have a full product.